python/dgl/data/tu.py

```diff
@@ -60,20 +70,60 @@ def __init__(self, name, use_pandas=False, hidden_size=10):
             one_hot_node_labels = self._to_onehot(DS_node_labels)
             for idxs, g in zip(node_idx_list, self.graph_lists):
                 g.ndata['feat'] = one_hot_node_labels[idxs, :]
+            self.data_mode = "node_label"
         except IOError:
             print("No Node Label Data")
 
         try:
             DS_node_attr = np.loadtxt(self._file_path("node_attributes"), delimiter=",")
             for idxs, g in zip(node_idx_list, self.graph_lists):
                 g.ndata['feat'] = DS_node_attr[idxs, :]
+            self.data_mode = "node_attr"
         except IOError:
             print("No Node Attribute Data")
 
         if 'feat' not in g.ndata.keys():
             for idxs, g in zip(node_idx_list, self.graph_lists):
                 g.ndata['feat'] = np.ones((g.number_of_nodes(), hidden_size))
+            self.data_mode = "constant"
             print("Use Constant one as Feature with hidden size {}".format(hidden_size))
+        
+        # remove graphs that are too large by user given standard
+        preserve_idx = []
+        print("original dataset length : ", len(self.graph_lists))
+        for (i, g) in enumerate(self.graph_lists):
+            if g.number_of_nodes() <= self.max_allow_node:
+                preserve_idx.append(i)
+        self.graph_lists = [self.graph_lists[i] for i in preserve_idx]
+        print("after pruning graphs that are too big : ", len(self.graph_lists))
+        self.graph_labels = [self.graph_labels[i] for i in preserve_idx]
+        self.max_num_node = self.max_allow_node
+        
+        # randomly shuffle the data
+        ind = [i for i in range(len(self.graph_labels))]
+        random.seed(321)
+        random.shuffle(ind)
+        self.graph_lists = [self.graph_lists[i] for i in ind]
+        self.graph_labels = [self.graph_labels[i] for i in ind]
+
+        #make dataset divisible by 10
+        if len(self.graph_lists) % 10 != 0:
+            print("TU dataset not divisible by 10")
+            new_len = len(self.graph_lists) // 10
+            self.graph_lists = self.graph_lists[:new_len*10]
+            self.graph_labels = self.graph_labels[:new_len*10]
+
+        #check dataset splitting:
+        eps = 1e-5
+        if split_ratio:
```

Comment:
Since you used Dataloader, I suggest move split and shuffle to the dataloader instead of dataset. The role of Dataset class should mainly be provide sample and corresponding label based on index (implementing `__getitem__` function). Shuffle and split should be done at Dataloader. You can refer to my implementation at https://github.com/VoVAllen/diffpool/blob/master/batched_train.py#L45 (don't follow the practice of moving data to gpu in collate function if you are using num_workers)